src/EditorFeatures/TestUtilities/Threading/WpfTheoryTestCase.cs

```diff
@@ -11,27 +11,17 @@ namespace Roslyn.Test.Utilities
 {
     public class WpfTheoryTestCase : XunitTheoryTestCase
     {
-        public WpfTestSharedData SharedData { get; private set; }
-
         [EditorBrowsable(EditorBrowsableState.Never)]
         [Obsolete("Called by the de-serializer; should only be called by deriving classes for de-serialization purposes")]
         public WpfTheoryTestCase() { }
 
         public WpfTheoryTestCase(IMessageSink diagnosticMessageSink, TestMethodDisplay defaultMethodDisplay, TestMethodDisplayOptions defaultMethodDisplayOptions, ITestMethod testMethod)
             : base(diagnosticMessageSink, defaultMethodDisplay, defaultMethodDisplayOptions, testMethod)
         {
-            SharedData = WpfTestSharedData.Instance;
         }
-
-        public override void Deserialize(IXunitSerializationInfo data)
-        {
-            base.Deserialize(data);
-            SharedData = WpfTestSharedData.Instance;
-        }
-
         public override Task<RunSummary> RunAsync(IMessageSink diagnosticMessageSink, IMessageBus messageBus, object[] constructorArguments, ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource)
         {
-            var runner = new WpfTheoryTestCaseRunner(SharedData, this, DisplayName, SkipReason, constructorArguments, diagnosticMessageSink, messageBus, aggregator, cancellationTokenSource);
+            var runner = new WpfTheoryTestCaseRunner(WpfTestSharedData.Instance, this, DisplayName, SkipReason, constructorArguments, diagnosticMessageSink, messageBus, aggregator, cancellationTokenSource);
```

Comment:
:bulb: Can we remove it from this constructor as well?

Reply:
@sharwell I can push it all the way down; not sure if this is a better dependency injection pattern? Otherwise you'll have classes directly using the static if anybody ever cares about that.